beaconChain/src/stateTransition/epoch/variables.ts

```diff
@@ -0,0 +1,111 @@
+import BN = require("bn.js");
+import {
+  getActiveValidatorIndices, getAttestationParticipants,
+  getBlockRoot, getCrosslinkCommitteesAtSlot, getCurrentEpoch, getEpochStartSlot, getPreviousEpoch, getTotalBalance,
+  slotToEpoch
+} from "../../helpers/stateTransitionHelpers";
+import {
+  Attestation, BeaconState, CrosslinkCommittee, Epoch, Gwei, PendingAttestation,
+  ValidatorIndex
+} from "../../types";
+
+export function processVariables(state: BeaconState) {
+  // Variables
+  const currentEpoch: Epoch = getCurrentEpoch(state);
+  const previousEpoch: Epoch = getPreviousEpoch(state);
+  const nextEpoch: Epoch = currentEpoch.addn(1);
+
+  // Validators attesting during the current epoch
+  const currentTotalBalance: Gwei = getTotalBalance(state, getActiveValidatorIndices(state.validatorRegistry, currentEpoch));
+  const currentEpochAttestations: PendingAttestation[] = state.latestAttestations.filter((attestation) => currentEpoch === slotToEpoch(attestation.data.slot));
+
+  // Validators justifying the epoch boundary block at the start of the current epoch
+  const currentEpochBoundaryAttestations: PendingAttestation[] = currentEpochAttestations.filter((attestation) => {
+    if (attestation.data.epochBoundaryRoot === getBlockRoot(state, getEpochStartSlot(currentEpoch))) {
+      return attestation;
+    }
+  });
+
+  const currentEpochBoundaryAttesterIndices: ValidatorIndex[] = currentEpochBoundaryAttestations
+    .map((attestation: PendingAttestation) => getAttestationParticipants(state, attestation.data, attestation.aggregationBitfield))
+    .reduce((previousValue: ValidatorIndex[], currentValue: ValidatorIndex[]) => previousValue.concat(...currentValue));
+
+  const currentEpochBoundaryAttestingBalance = getTotalBalance(state, currentEpochBoundaryAttesterIndices);
+
+  // Validators attesting during the previous epoch
+  const previousTotalBalance = getTotalBalance(state, getActiveValidatorIndices(state.validatorRegistry, previousEpoch));
```

Comment:
add `Gwei` type declaration here for consistency

Reply:
done